### PR TITLE
Refactor document printing

### PR DESCRIPTION
**Pull request by ericdill**
Recursively print the Document class and format the DataKeys of an EventDescriptor into a table. 
## Example header output

``` python
from dataportal import DataBroker as db
print(db[-1])
```

produces

```
Header
------
beamline_id    : csx                                     
group          :                                         
moon           : full                                    
owner          :                                         
plotx          : linear_motor                            
ploty          : [u'total_img_sum']                      
project        :                                         
run_start_id   : 5538f65f3892c9066bf8917a                
run_start_uid  : 5a84b162-126b-449b-b00a-c2b506c9619e    
sample         : {}                                      
scan_id        : 36                                      
start_datetime : 2015-04-23 09:40:47.857965              
start_time     : 1429796447.86                           

  BeamlineConfig
  ==============
  config_params  : {u'my_beamline': u'my_value'}           
  id             : 5538f65f3892c9066bf89179                
  time           : 1429796447.84                           
  time_as_datetim: 2015-04-23 09:40:47.835954              
  uid            : 8990e097-000c-4592-841d-3dcdfc04bb88    

  EventDescriptor
  ===============
  +---------------+--------+------------+----------+-------------+
  | key name      | dtype  |   shape    | external |    source   |
  +---------------+--------+------------+----------+-------------+
  | img           | array  | FILESTORE: |  [5, 5]  |     CCD     |
  | img_sum_x     | array  | FILESTORE: |   [5]    |   CCD:xsum  |
  | img_sum_y     | array  | FILESTORE: |   [5]    |   CCD:ysum  |
  | img_x_max     | number |    None    |    []    |   CCD:xmax  |
  | img_y_max     | number |    None    |    []    |   CCD:ymax  |
  | linear_motor  | number |    None    |    []    | PV:ES:sam_x |
  | total_img_sum | number |    None    |    []    |   CCD:sum   |
  +---------------+--------+------------+----------+-------------+
  id             : 5538f65f3892c9066bf8917b                
  time           : 0.0                                     
  time_as_datetim: 1969-12-31 19:00:00                     
  uid            : 5f6559a2-f643-4ce8-9615-43538ddf0e71    


  EventDescriptor
  ===============
  +----------+--------+-------+----------+------------+
  | key name | dtype  | shape | external |   source   |
  +----------+--------+-------+----------+------------+
  | Tsam     | number |  None |    []    | PV:ES:Tsam |
  +----------+--------+-------+----------+------------+
  id             : 5538f65f3892c9066bf8917c                
  time           : 0.0                                     
  time_as_datetim: 1969-12-31 19:00:00                     
  uid            : 2aa492c6-072d-4de0-a5ea-8c87dffcdb4f    
```

### CODE REVIEW

**Comment by tacaswell** [comment]
Can you add prettytable to  conda `meta.yaml`,  `setup.py`, and to `.travis.yml`?

**Comment by ericdill** [comment]
Where would I add prettytable to setup.py?

File: .travis.yml [hunk 1]
```diff
@@ -1,36 +1,36 @@
-language: python
```
Comment (tacaswell) [inline .travis.yml:1]: Why is every line in this file changed?

**Comment by tacaswell** [comment]
ah, it does not have the section where it would go, skip that then.

File: .travis.yml (hunk 1 above)
Comment (ericdill) [inline .travis.yml:1]: presumably because they were crlf endings

**Comment by danielballan** [comment]
I like this a lot. Can one more character be allocated so that `time_as_datetim` can have its `e`?

**Comment by danielballan** [comment]
Bummer:

```
first_event.descriptor.data_keys
AttributeError: 'Document' object has no attribute 'uid'
```

**Pull request merged**